### PR TITLE
Add ramdump support for stm32l4

### PR DESCRIPTION
**Pull request by thapav**
This patch enables ramdump support using TizenRT ramdump tool for stm32l4 board.

#### RESULT is as follows:

Please enter serial port adapter:
For example: /dev/ttyUSB1 or /dev/ttyACM0
Enter:
/dev/ttyACM0
[sudo] password for vidisha:
Target Handshake successful do_handshake
Target entered to ramdump mode

=========================================================================
Ramdump Region Options:
1. ALL
2. Region 0:    ( Address: 0x20000000, Size: 196608 )    [Heap index = 0]
3. Region 1:    ( Address: 0x20030000, Size: 65536 )     [Heap index = 0]
4. Region 2:    ( Address: 0x20040000, Size: 393216 )    [Heap index = 0]
=========================================================================
Please enter desired ramdump option as below:
        1 for ALL
        2 for Region 0
        25 for Region 0 & 3 ...
Please enter your input: 1

No. of Regions to be dumped received: ramdump_recv

Receiving ramdump......

********************************************************************************
Dumping Region: 0, Address: 0x20000000, Size: 196608bytes
********************************************************************************
[============>]
********************************************************************************
Dumping Region: 1, Address: 0x20030000, Size: 65536bytes
********************************************************************************
[====>]
********************************************************************************
Dumping Region: 2, Address: 0x20040000, Size: 393216bytes
********************************************************************************[========================>]
Ramdump received successfully..!

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [4e521fa5271321f3b3c1fa0db6d63b73841067f1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4e521fa5271321f3b3c1fa0db6d63b73841067f1] - Code Rule Check OK.

**Comment by thapav** [comment]
@sunghan-chang  @sijupunnnoose  Kindly review

File: os/board/stm32l4r9ai-disco/src/stm32_crashdump.c [hunk 1]
```diff
@@ -0,0 +1,219 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
```
Comment (amandeepc) [inline os/board/stm32l4r9ai-disco/src/stm32_crashdump.c]: 2017 -> 2019

File: os/board/stm32l4r9ai-disco/src/Makefile [hunk 2]
```diff
@@ -89,6 +89,13 @@ VPATH += :$(TOPDIR)/board/common
 CSRCS += partitions.c
 endif
 
+DEPPATH += --dep-path $(TOPDIR)/board/common
+VPATH += :$(TOPDIR)/board/common
+
+ifeq ($(CONFIG_BOARD_CRASHDUMP),y)
+CSRCS += stm32_crashdump.c
```
Comment (sunghan-chang) [inline os/board/stm32l4r9ai-disco/src/Makefile]: 1. What is the difference between crashdump.c and stm32_crashdump.c? Could you let me know why it is necessary newly?
2. If this is necessary, you should not add it in common. You should put it in stm folder becaues it is not common.

File: os/arch/arm/src/stm32l4/hal/stm32l4xx_hal_interface.c [hunk 3]
```diff
@@ -1,233 +1,244 @@
...
-/**
-  * @}
-  */
-
-/**
-  * @}
-  */
-
-/************************ (C) COPYRIGHT STMicroelectronics *****END OF FILE****/
+/**
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32l4/hal/stm32l4xx_hal_interface.c]: Could you check this file? It is something wrong.

File: os/board/stm32l4r9ai-disco/src/Makefile (hunk 2 above)
Comment (thapav) [reply os/board/stm32l4r9ai-disco/src/Makefile]: The low level API's used in stm32_crashdump.c are implemented in hal layer (stm32l4xx_hal_interface.c ), there are separate up_lowptc() & up_getc() implementations in stm32l4_serial.c
If we wish to override these implementations in serial file with the ones in stm32l4xx_hal_interface.c , then we can move it to common file crashdump.c else we will have to keep it separate.
What is your opinion ?

Comment (sunghan-chang) [reply os/board/stm32l4r9ai-disco/src/Makefile]: File name is not important. Anyway stm also provides up_lowputc and up_getc in arch as others do. Then there is no problem to use common. It looks the same as common.

Comment (thapav) [reply os/board/stm32l4r9ai-disco/src/Makefile]: Ramdump fails with the up_lowptc() & up_getc() implementations in stm32l4_serial.c.
So shall I go ahead and modify them ?

**Comment by seinfra** [comment]
Target : [bd1a7c1e497bd6728d606f59234753cfcbd6ee9a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bd1a7c1e497bd6728d606f59234753cfcbd6ee9a] - Code Rule Check OK.

**Comment by thapav** [comment]
Updated the patch, please check now

**Pull request merged**